### PR TITLE
Refactor integration tests with FileFollower

### PR DESCRIPTION
**Pull request by mkemel**
This PR has two commits:

- refactor bluechictl-monitor-unit-created-removed
- refactor bluechi-monitor-service test

This is a refactoring for these two tests to use the new BluechiCtl BackgroundRunner API

### CODE REVIEW

**Review by mwperina — APPROVED** [review]
LGTM

File: tests/bluechi_machine_lib/bluechictl.py [hunk 1]
```diff
@@ -119,23 +119,47 @@ def stop(self):
 
 
 class BluechiCtl:
```
Comment (engelmi) [inline tests/bluechi_machine_lib/bluechictl.py:121]: I am not sure we should keep two implementations of `BluechiCtl`. I think moving the `bluechi_test.bluechictl` to `bluechi_machine_lib` and extending it with the functions of the current `bluechi_machine_lib` would be best. The `subprocess.Popen` can be encapsulated in another `Client` implementation, e.g. `ProcessClient`.

Not in this PR, but would be great to do this in a follow-up.

File: tests/tests/tier0/bluechi-monitor-service/python/monitor.py [hunk 2]
```diff
@@ -1,122 +1,56 @@
 # SPDX-License-Identifier: LGPL-2.1-or-later
```
Comment (engelmi) [inline tests/tests/tier0/bluechi-monitor-service/python/monitor.py:1]: Looking at the refactored code, the `python/monitor.py` wouldn't be necessary - everything could be done in the `test_bluechictl_monitor_xyz`.

Comment (mkemel) [reply tests/tests/tier0/bluechi-monitor-service/python/monitor.py:1]: How? We need the FileFollower and all that stuff should run in the container itself

File: tests/bluechi_machine_lib/bluechictl.py (hunk 1 above)
Comment (mkemel) [reply tests/bluechi_machine_lib/bluechictl.py:121]: But these are two different things.
One is running locally. The other runs with `exec`

Comment (engelmi) [reply tests/bluechi_machine_lib/bluechictl.py:121]: `exec_run` is defined on the `Client` interface. So having a Client-subclass that implements it using the `subprocess.popen` (or uses the `Command` class for it) encapsulates that.

Comment (mkemel) [reply tests/bluechi_machine_lib/bluechictl.py:121]: But this way you will copy it to `bluechi_machine_lib`, and then you are getting code that you can't run on the machine - on the machine

File: tests/tests/tier0/bluechi-monitor-service/python/monitor.py (hunk 2 above)
Comment (engelmi) [reply tests/tests/tier0/bluechi-monitor-service/python/monitor.py:1]: Hmm... the FileFollower isn't a problem, but the BackgroundRunner is due to the `subprocess.Popen`. Lets keep it as it is for now.

File: tests/bluechi_machine_lib/bluechictl.py (hunk 1 above)
Comment (engelmi) [reply tests/bluechi_machine_lib/bluechictl.py:121]: But as I wrote: maybe as a follow-up :)

**Review by engelmi — APPROVED** [review]

**Pull request merged**